extensions/http/src/main/java/org/mule/extension/http/api/listener/HttpListener.java

```diff
@@ -0,0 +1,415 @@
+import org.mule.runtime.core.config.i18n.CoreMessages;
+import org.mule.runtime.extension.api.annotation.Alias;
+import org.mule.runtime.extension.api.annotation.Expression;
+import org.mule.runtime.extension.api.annotation.Parameter;
+import org.mule.runtime.extension.api.annotation.param.Connection;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.extension.api.annotation.param.UseConfig;
+import org.mule.runtime.extension.api.runtime.source.Source;
+import org.mule.runtime.module.http.api.HttpConstants;
+import org.mule.runtime.module.http.internal.HttpParser;
+import org.mule.runtime.module.http.internal.domain.ByteArrayHttpEntity;
+import org.mule.runtime.module.http.internal.domain.EmptyHttpEntity;
+import org.mule.runtime.module.http.internal.domain.HttpProtocol;
+import org.mule.runtime.module.http.internal.domain.request.HttpRequestContext;
+import org.mule.runtime.module.http.internal.domain.response.DefaultHttpResponse;
+import org.mule.runtime.module.http.internal.domain.response.HttpResponse;
+import org.mule.runtime.module.http.internal.domain.response.ResponseStatus;
+import org.mule.runtime.module.http.internal.listener.HttpRequestParsingException;
+import org.mule.runtime.module.http.internal.listener.HttpThrottlingHeadersMapBuilder;
+import org.mule.runtime.module.http.internal.listener.ListenerPath;
+import org.mule.runtime.module.http.internal.listener.RequestHandlerManager;
+import org.mule.runtime.module.http.internal.listener.Server;
+import org.mule.runtime.module.http.internal.listener.async.HttpResponseReadyCallback;
+import org.mule.runtime.module.http.internal.listener.async.RequestHandler;
+import org.mule.runtime.module.http.internal.listener.async.ResponseStatusCallback;
+import org.mule.runtime.module.http.internal.listener.matcher.AcceptsAllMethodsRequestMatcher;
+import org.mule.runtime.module.http.internal.listener.matcher.ListenerRequestMatcher;
+import org.mule.runtime.module.http.internal.listener.matcher.MethodRequestMatcher;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.apache.commons.collections.map.MultiValueMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Alias("listener")
+public class HttpListener extends Source<Object, HttpRequestAttributes> implements Initialisable
+{
+    private static final Logger logger = LoggerFactory.getLogger(HttpListener.class);
+    private static final String SERVER_PROBLEM = "Server encountered a problem";
+
+    @UseConfig
+    private HttpListenerConfig config;
+
+    @Connection
+    private Server server;
+
+    @Parameter
+    @Expression(NOT_SUPPORTED)
+    private String path;
+
+    /**
+     * Comma separated list of allowed HTTP methods by this listener. To allow all methods do not defined the attribute.
+     */
+    @Parameter
+    @Optional
+    @Expression(NOT_SUPPORTED)
+    private String allowedMethods;
```

Comment:
for now let's leave it, but consider if it's not better to expose this a Set of a method's enum